### DISCUSSION TITLE
Possible issue in the code

### DISCUSSION THREAD

**Discussion by ns-ggeorgiev** [renovatebot/renovate#17214] [Suggest an Idea]
I noticed the following in the code:
```
    branches.sort((a, b) => {
        if (a.isVulnerabilityAlert && !b.isVulnerabilityAlert) {
            return -1;
        }
        ...
```
most sort algorithms expect the predicate function to be symmetrical, else it might get in endless loop, crash or generate random result.

With other words it will be safer if the code looks like these:

```
    branches.sort((a, b) => {
        if (a.isVulnerabilityAlert && !b.isVulnerabilityAlert) {
            return -1;
        }
        if (!a.isVulnerabilityAlert && b.isVulnerabilityAlert) {
            return 1;
        }
        ...
```

**Comment by viceice** [comment]
feel free to open a pr to fix that. Did yo see any issues?

**Reply by ns-ggeorgiev** [reply]
I haven't. I do not have vulnerabilityAlert enabled. I discovered that with a code review.

**Reply by HonkingGoose** [reply]
I created an issue to track this work, see:

- #17276
